packages/lib/src/components/Login/components/LoginForm/LoginForm.test.tsx

```diff
@@ -17,56 +17,119 @@
  * along with Essencium Frontend. If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { fireEvent, render, RenderResult, screen } from '@testing-library/react'
+import { AppShell, MantineProvider } from '@mantine/core'
+import {
+  cleanup,
+  fireEvent,
+  render,
+  RenderResult,
+  screen,
+} from '@testing-library/react'
 import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
 
 import { LoginForm } from './LoginForm'
 
 describe('LoginForm', () => {
   let LoginFormMounted: RenderResult
 
-  let emailInput: HTMLInputElement
-  let passwordInput: HTMLInputElement
-  let submitButton: HTMLButtonElement
-
-  const handlePasswordReset = vi.fn()
   const setIsPasswordResetFormOpened = vi.fn()
-  const isPasswordResetFormOpened = false
+  const handlePasswordReset = vi.fn()
+  const handleLogin = vi.fn()
+
   const isResetPasswordSent = false
   const isResettingPassword = false
 
   beforeEach(() => {
+    const isPasswordResetFormOpened = false
+
     LoginFormMounted = render(
-      <LoginForm
-        handleLogin={() => {}}
-        handlePasswordReset={handlePasswordReset}
-        setIsPasswordResetFormOpened={setIsPasswordResetFormOpened}
-        isResetPasswordSent={isResetPasswordSent}
-        isPasswordResetFormOpened={isPasswordResetFormOpened}
-        isResettingPassword={isResettingPassword}
-      />,
+      <MantineProvider>
+        <AppShell>
+          <LoginForm
+            handleLogin={handleLogin}
+            handlePasswordReset={handlePasswordReset}
+            setIsPasswordResetFormOpened={setIsPasswordResetFormOpened}
+            isResetPasswordSent={isResetPasswordSent}
+            isPasswordResetFormOpened={isPasswordResetFormOpened}
+            isResettingPassword={isResettingPassword}
+          />
+          ,
+        </AppShell>
+      </MantineProvider>,
     )
   })
 
-  afterEach(() => {
+  afterEach(cleanup)
+
+  it('sohuld render email and password inputs as well as the login button and password reset link', () => {
+    expect(
+      screen.getByRole('textbox', { name: 'loginView.form.email' }),
+    ).toBeDefined()
+
+    expect(screen.getByLabelText('loginView.form.password')).toBeDefined()
```

Comment:
Why do you get one by role and the other by label text?

Reply:
This is due to the HTML spec, input fields with type password do not have a role. [Here](https://github.com/testing-library/dom-testing-library/issues/567#issue-616906804) you can read more about it. But I will leave a comment inside the code, I forgot it, sorry!